pkg/wait/wait_for_ready.go

```diff
@@ -111,20 +111,20 @@ func NoopMessageCallback() MessageCallback {
 func (w *waitForReadyConfig) Wait(watcher watch.Interface, name string, options Options, msgCallback MessageCallback) (error, time.Duration) {
 
 	timeout := options.timeoutWithDefault()
-	floatingTimeout := timeout
+	timeoutTimer := time.NewTimer(timeout)
 	for {
 		start := time.Now()
-		retry, timeoutReached, err := w.waitForReadyCondition(watcher, start, floatingTimeout, options.errorWindowWithDefault(), msgCallback)
+		retry, timeoutReached, err := w.waitForReadyCondition(watcher, start, timeoutTimer, options.errorWindowWithDefault(), msgCallback)
 		if err != nil {
 			return err, time.Since(start)
 		}
-		floatingTimeout = floatingTimeout - time.Since(start)
-		if timeoutReached || floatingTimeout < 0 {
+		if timeoutReached {
 			return fmt.Errorf("timeout: %s '%s' not ready after %d seconds", w.kind, name, int(timeout/time.Second)), time.Since(start)
 		}
 
 		if retry {
-			// restart loop
+			// sleep to prevent CPU pegging and restart the loop
+			time.Sleep(pollInterval)
```

Comment:
This isn't necessary if there's no longer a chance the CPU can be pegged. However, unless you're 100% sure that can't happen, then it should remain.

Reply:
This is not really clear to me. As I understand this, it's meant as a safeguard in case the retry loop would go wild again (like that `waitForReadyCondition` returning immediately with retry == true, which has happened up to now when not detecting that the watcher channel has been closed). 

Looking at the code I can't see how the CPU could be pegged, but I have no prove. Adding a sleep here however would make it hard to detect such a situation, but of course, could be 'good enough'. So I'm a bit biased here :)

However,  I would not couple this to the `pollInterval` which is used as poll interval for a special kind of watcher (i.e. when the API does not support intrinsic watching like this is the case for CloudRun). Instead, I probably would use a shorter safe-guard, maybe 200 or 500ms ? (but I would use a different constant or even a literal value here).